datafusion/common/src/scalar.rs

```diff
@@ -1562,7 +1597,11 @@ impl ScalarValue {
                     DataType::UInt16 => dict_from_values::<UInt16Type>(values)?,
                     DataType::UInt32 => dict_from_values::<UInt32Type>(values)?,
                     DataType::UInt64 => dict_from_values::<UInt64Type>(values)?,
-                    _ => unreachable!("Invalid dictionary keys type: {:?}", key_type),
```

Comment:
I agree it makes more sense to keep `unreachable!` here to align with how `arrow-rs` deals with illegal dictionary key types.